### PR TITLE
Reduce odds of race in scrollUntilVisible

### PR DESCRIPTION
**Pull request by cbracken**
In FlutterDriver.scrollUntilVisible, pause momentarily after we kick off
the waitFor() call, in order to give that call a chance to complete
before we start scrolling if the element is on-screen.

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]
/cc @HansMuller, the original author of this method.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -480,19 +480,11 @@ class FlutterDriver {
...
-      // Assume that that waitFor timed out because the item isn't visible.
-    }
-
-    if (!isVisible) {
-      waitFor(item, timeout: timeout).then((Null _) { isVisible = true; });
-      while (!isVisible) {
-        await scroll(scrollable, dxScroll, dyScroll, const Duration(milliseconds: 100));
-        await new Future<Null>.delayed(const Duration(milliseconds: 500));
-      }
+    waitFor(item, timeout: timeout).then((Null _) { isVisible = true; });
```
Comment (HansMuller) [inline packages/flutter_driver/lib/src/driver/driver.dart]: I'll accept all the blame for the implicit rationale for these timeouts and delays. Please add some comments that explain exactly what's going on here, for the sake of the next person who's stuck with trying to understand what's going on...

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 2]
```diff
@@ -480,19 +480,11 @@ class FlutterDriver {
 
     // If the item is already visible then we're done.
     bool isVisible = false;
-    try {
```
Comment (HansMuller) [inline packages/flutter_driver/lib/src/driver/driver.dart]: Does the new version of the code really work when the item isn't initially visible? The DriverError test here used to be needed, what changed?

**Review by HansMuller — COMMENTED** [review]
LGTM

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 1 above)
Comment (cbracken) [reply packages/flutter_driver/lib/src/driver/driver.dart]: Good idea -- done!

**Pull request merged**